lib/studies/reply/adaptive_nav.dart

```diff
@@ -701,92 +704,95 @@ class _MobileNavState extends State<_MobileNav> with TickerProviderStateMixin {
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      extendBody: true,
-      body: LayoutBuilder(
-        builder: _buildStack,
-      ),
-      bottomNavigationBar: Consumer<EmailStore>(
-        builder: (context, model, child) {
-          _bottomAppBarController.forward();
-
-          return SizeTransition(
-            sizeFactor: _bottomAppBarCurve,
-            axisAlignment: -1,
-            child: BottomAppBar(
-              shape: const CircularNotchedRectangle(),
-              notchMargin: 8,
-              child: SizedBox(
-                height: kToolbarHeight,
-                child: Row(
-                  mainAxisSize: MainAxisSize.max,
-                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
-                  children: [
-                    InkWell(
-                      borderRadius: const BorderRadius.all(Radius.circular(16)),
-                      onTap: _toggleBottomDrawerVisibility,
-                      child: Row(
-                        children: [
-                          const SizedBox(width: 16),
-                          RotationTransition(
-                            turns: Tween(
-                              begin: 0.0,
-                              end: 1.0,
-                            ).animate(_dropArrowCurve),
-                            child: const Icon(
-                              Icons.arrow_drop_up,
-                              color: ReplyColors.white50,
+    return _SharedAxisTransitionSwitcher(
+      defaultChild: Scaffold(
+        extendBody: true,
+        body: LayoutBuilder(
+          builder: _buildStack,
+        ),
+        bottomNavigationBar: Consumer<EmailStore>(
+          builder: (context, model, child) {
+            _bottomAppBarController.forward();
```

Comment:
Why does the bottomAppBarController need to go forward here?

Reply:
It goes forward here to move the bottom app bar back into view, once we navigate to the `MailViewPage`. This is done in case we navigate to an Email and our bottom app bar was hidden, which if the email wasn't scrollable would give us no way to get the bottom app bar into view.